### ISSUE TITLE
Udc - add/set/remove Quantity + revive

### ISSUE THREAD

**Issue by samueleceroni** [Zeegomo/nwoolcan#13]
I think there are no use cases for these functions. A Udc should have its own quantity when it is created, and it can only be consumed, "removing" some of (or all) its quantity. Instead of adding a Quantity to the Udc you should create a new Udc with a new Quantity. This is the only way to track the life cycle of the articles. It follows that there is no use case for setting a udc quantity.
Furthermore, regarding articles which are consumed and which could be used again at a later point (some of misc articles e.g. bottles), I would prefer to have a function which "revives" those articles (reporting errors where not possible).

**Comment by Zeegomo** [comment]
i don't get what "revive" should mean but it is ok if udc does not have an add method

**Comment by samueleceroni** [comment]
I will give an example to clarify 'revive'.
You have n bottles and you use them.
'available' -> 'consumed'.
now you drink 3 beers and you get back 3 bottles. you 'revive' 3 bottles.
'consumed' -> 'available'.

of course this method can't be applied to every article (you can't revive hops lol).

**Comment by GiackAloZ** [comment]
There is no such thing as reviving since creating a new Udc/adding to an existing one is better in this case.
Anyway, consumed items are only suppose to be of type beer, so there is non such thing as a consumed bottle.
Furthermore, the add, remove methods of udc are just intended to be used to keep track of the quantity of things like bottles or caps (that dont need to be tracked in a "consumed" way).
Beers should be tracked (so consumed) instead.

In conclusion, i think that add/set/remove functionalities should be there as previously decided.

**Comment by samueleceroni** [comment]
Actually I don't agree with this position. I think that we should try to generalize the concept to all type of article, or it will easily become a mess.
I agree that the term 'consumed' is not that good for bottles or caps. We should consider some other state instead of that. What about 'used'?
I think that the 'revive' approach is better because it actually is what bottles do. They are used and then revived, ready to be used again, but those are not different bottles (while you can't revive hops because you'll never use the same hop again). Do you get my point?

**Comment by GiackAloZ** [comment]
You can actually find yourself a solution for this problem:
- we need to have the reference of the batch in a beer article (that can be reached also by the UI)
- we need to consume a beer, but keep track of this consumption
- we need to have a distinction between ingredients and misc stuff
- we need to add or remove quantities of ingredients and misc stuff
- we need to be able to create a new beer at the end of a batch production

So, if you find a better solution to this problem, you can just use that instead.
Afterall, is your contribuiton to this project to also find a solution that fits best to you.

**Comment by samueleceroni** [comment]
I was looking for a constructive argumentation of the problem, I was not just criticizing it because I enjoy to do it. BTW I will think about it by myself and I'll let you know.
